referenceframe/frame.go

```diff
@@ -445,3 +445,133 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(temp)
 }
+
+type poseFrame struct {
+	*baseFrame
+	geometries []spatial.Geometry
+}
+
+// NewPoseFrame creates an orientation vector frame, i.e a frame with
+// 7 degrees of freedom: X, Y, Z, OX, OY, OZ, and Theta in radians.
+func NewPoseFrame(name string, geometry []spatial.Geometry) (Frame, error) {
+	orientationVector := spatial.OrientationVector{
```

Comment:
> You can just construct the limits for the orientation by hand they should be betweeen -1 and 1 because that is the domain of something that has been normalized.

Not necessarily true, we can have `OZ: 1.000000004` this resolves that issue.

Reply:
extra context: https://github.com/viamrobotics/rdk/pull/3876#discussion_r1638931323

Reply:
This is ensuring that the limits of the orientation vector are whatever the unit vector is since it will always be the same after you normalize it.  Why not just use +/- infinity for them instead?

Reply:
I am confused now.
I can change the limits to be 
```
limits := []Limit{
	{Min: math.Inf(-1), Max: math.Inf(1)}, // X
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Y
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Z
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OX
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OY
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OZ
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Theta
}
```
opposed to 
```
orientationVector := spatial.OrientationVector{
	OX:    math.Inf(1),
	OY:    math.Inf(1),
	OZ:    math.Inf(1),
	Theta: math.Inf(1),
}
orientationVector.Normalize()
limits := []Limit{
	{Min: math.Inf(-1), Max: math.Inf(1)},                         // X
	{Min: math.Inf(-1), Max: math.Inf(1)},                         // Y
	{Min: math.Inf(-1), Max: math.Inf(1)},                         // Z
	{Min: -orientationVector.OX, Max: orientationVector.OX},       // OX
	{Min: -orientationVector.OY, Max: orientationVector.OY},       // OY
	{Min: -orientationVector.OZ, Max: orientationVector.OZ},       // OZ
	{Min: -orientationVector.Theta, Max: orientationVector.Theta}, // Theta
}
```
but seems to be going against the ask of this comment here https://github.com/viamrobotics/rdk/pull/3876#discussion_r1638931323

Reply:
We can't keep it this way.  Normalizing the orientation vector this way returns NaNs then you set the min and max to be positive and negative NaN.

Reply:
I must have missed this on an earlier review; what is done here is not what I meant in the comment linked above, apologies for miscommunicating.

In the comment above I was referring to the normalization that would automatically happen on the backend with a real pose, but rereading my own comment it does look like I was asking for the normalization to happen here.

Ray is correct. The limits should be a simple 
```
limits := []Limit{
	{Min: math.Inf(-1), Max: math.Inf(1)}, // X
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Y
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Z
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OX
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OY
	{Min: math.Inf(-1), Max: math.Inf(1)}, // OZ
	{Min: math.Inf(-1), Max: math.Inf(1)}, // Theta
}
```